pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -304,8 +302,7 @@ private void addInstanceGroupTagIfNeeded() {
    * @return Helix instance config
    */
   public InstanceConfig getHelixInstanceConfig(@Nonnull String instanceId) {
-    ZNRecord znRecord = _cacheInstanceConfigsDataAccessor.get("/" + instanceId, null, AccessOption.PERSISTENT);
-    return znRecord != null ? new InstanceConfig(znRecord) : null;
+    return _helixAdmin.getInstanceConfig(_helixClusterName, instanceId);
```

Comment:
This will throw exception when instance does not exist. You might want to use `_helixDataAccessor.getProperty(_keyBuilder.instanceConfig(instanceId))`

Reply:
Updated. 👍